lib/scala/runtime-version-manager/src/main/scala/org/enso/runtimeversionmanager/runner/JavaExecCommand.scala

```diff
@@ -0,0 +1,69 @@
+package org.enso.runtimeversionmanager.runner
+
+import org.enso.runtimeversionmanager.components.{Engine, GraalRuntime}
+import org.enso.runtimeversionmanager.components.Manifest.JVMOptionsContext
+
+/** Represents a way of launching the JVM.
+  *
+  * @param executableName name of the `java` executable to run
+  * @param javaHomeOverride if set, asks to override the JAVA_HOME environment
+  *                         variable when launching the JVM
+  */
+class JavaExecCommand(
+  val executableName: String,
+  val javaHome: Option[String]
+) extends ExecCommand {
+  def path: String = executableName
+
+  override def cmdArguments(
+    engine: Engine,
+    jvmSettings: JVMSettings
+  ): Seq[String] = {
+    def translateJVMOption(
+      option: (String, String),
+      standardOption: Boolean
+    ): String = {
+      val name  = option._1
+      val value = option._2
+      if (standardOption) s"-D$name=$value" else s"--$name=$value"
+    }
+
+    val context = JVMOptionsContext(enginePackagePath = engine.path)
+
+    val manifestOptions =
+      engine.defaultJVMOptions
+        .filter(_.isRelevant)
+        .map(_.substitute(context))
+    val commandLineOptions = jvmSettings.jvmOptions.map(
+      translateJVMOption(_, standardOption = true)
+    ) ++ jvmSettings.extraOptions.map(
+      translateJVMOption(_, standardOption = false)
+    )
+    val componentPath = engine.componentDirPath.toAbsolutePath.normalize
+    val modulePathOptions =
+      Seq(
+        "--module-path",
+        componentPath.toString,
+        "-m",
+        "org.enso.runner/org.enso.runner.Main"
```

Comment:
Rather referencing `org.enso.runner.Main` class, the `project-manager` should just use `NativeExecCommand` and pass in `--jvm` - once #12302 is fixed.